File: maxhanna.client/src/services/datacontracts/nexus/nexus-unit-upgrades.ts
export class NexusUnitUpgrades {
  id: number;
  coordsX: number;
  coordsY: number;
  unitIdUpgraded: number;
  timestamp: Date;

  constructor(id: number, coordsX: number, coordsY: number, unitIdUpgraded: number, timestamp: Date) {
    this.id = id;
    this.coordsX = coordsX;
    this.coordsY = coordsY;
    this.unitIdUpgraded = unitIdUpgraded;
    this.timestamp = timestamp;
  }
}

File: maxhanna.Server/Controllers/DataContracts/Nexus/NexusUnitUpgrades.cs
namespace maxhanna.Server.Controllers.DataContracts.Nexus
{
	public class NexusUnitUpgrades
	{
		public int Id { get; set; }
		public int CoordsX { get; set; }
		public int CoordsY { get; set; }
		public int UnitIdUpgraded { get; set; }
		public DateTime Timestamp { get; set; }

		public NexusUnitUpgrades()
		{
		}

		public NexusUnitUpgrades(int id, int coordsX, int coordsY, int unitIdUpgraded, DateTime timestamp)
		{
			Id = id;
			CoordsX = coordsX;
			CoordsY = coordsY;
			UnitIdUpgraded = unitIdUpgraded;
			Timestamp = timestamp;
		}
	}

}
